src/Sulu/Component/Content/Types/BlockContentType.php

```diff
@@ -374,32 +386,9 @@ private function prepareData(PropertyInterface $property, callable $dataCallback
             $blockPropertyType = $blockProperty->getProperties($i);
             $blockPropertyTypeSettings = $blockPropertyType->getSettings();
 
-            if (
-                \is_array($blockPropertyTypeSettings)
-                && !empty($blockPropertyTypeSettings['hidden'])
-            ) {
-                continue;
-            }
-
-            if (\is_array($blockPropertyTypeSettings)) {
-                $webspaceKey = $this->requestAnalyzer->getWebspace()->getKey();
-                $segment = $this->requestAnalyzer->getSegment();
-                if (isset($blockPropertyTypeSettings['segment_enabled'])
-                    && $blockPropertyTypeSettings['segment_enabled']
-                    && isset($blockPropertyTypeSettings['segments'][$webspaceKey])
-                    && $segment
-                    && $blockPropertyTypeSettings['segments'][$webspaceKey] !== $segment->getKey()
-                ) {
-                    continue;
-                }
-
-                if (isset($blockPropertyTypeSettings['target_groups_enabled'])
-                    && $blockPropertyTypeSettings['target_groups_enabled']
-                    && isset($blockPropertyTypeSettings['target_groups'])
-                    && $this->targetGroupStore
-                    && !\in_array($this->targetGroupStore->getTargetGroupId(), $blockPropertyTypeSettings['target_groups'])
-                ) {
-                    continue;
+            foreach ($this->blockSkippers as $blockSkipper) {
```

Comment:
would it be possible to extract this functionality and include it into a `2.1.*` release? that would really simplify supporting this in the `SuluHeadlessBundle` 🙂

Reply:
I think I would not do it... It's a new feature, and therefore should not be released in a bugfix release IMO. Also, what is there to support in 2.1 regarding this? IIRC in Sulu 2.1 the UI controls for defining this stuff did not even exist for the functionality being implemented in these interfaces, so I would say it is safe to just not register these services if this interface does not exist.

Reply:
I am sorry - you are right. I was thinking about a `2.2.*` release, not a `2.1.*` release 🙈  I adjusted the comment above.

Reply:
Right, some of this hiding functionality is already in Sulu 2.2, forgot about that :see_no_evil: Personally I would still prefer to merge it only in master, to avoid changing too much in existing releases... Apart from that the interface will change once more to be something like a `Visitor` anyway...

I think I would definitely go to merge everything in this PR, and hopefully this is tested somewhere, so that we see if everything works as expected. When you then try to implement it in the SuluHeadlessBundle, I would try to go first without requiring a change in sulu/sulu, and if it is really that much of a hassle we can still backport that part.